src/Storages/ObjectStorageQueue/ObjectStorageQueueIFileMetadata.cpp

```diff
@@ -344,20 +345,25 @@ void ObjectStorageQueueIFileMetadata::setFailedNonRetriable()
         return;
     }
 
-    if (Coordination::isHardwareError(responses[0]->error))
-    {
-        LOG_WARNING(log, "Cannot set file as failed: lost connection to keeper");
-        return;
-    }
+    if (Coordination::isHardwareError(code))
+        throw zkutil::KeeperMultiException(code, requests, responses);
 
-    if (responses[0]->error == Coordination::Error::ZNODEEXISTS)
```

Comment:
this check is removed, because it was anyway a logical error (there was `chassert(false)`), so ok to fallback to exception below instead.